tests/backup/backup_rancher_test.go

```diff
@@ -265,3 +266,211 @@ var _ = Describe("{SingleNamespaceBackupRestoreToNamespaceInSameAndDifferentProj
+		})
+
+		Step("Restore the backup taken with replace policy and move destination ns from one project to another project while restoring", func() {
+			log.InfoD("Restore the backup taken with replace policy and move ns from one project to another project while restoring")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreName = fmt.Sprintf("%s-replace-ns-project-move-%v", restoreNamePrefix, RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreOnRancherWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx, nil, projectUIDMapping, projectNameMapping, 2)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore- %s with replace policy from backup %s", restoreName, backupName))
+			log.Infof("Move namespaces in destination cluster to diff project while restore is going on")
+			err = Inst().S.(*rke.Rancher).ChangeProjectForNamespace(destProjectList[1], restoreNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Moving namespaces %v from project %s to project %s", restoreNamespaceList, destProjectList[0], destProjectList[1]))
+			log.Infof("Verifying if restore is successful after moving destination namespaces from one project to another")
+			err = restoreSuccessCheck(restoreName, orgID, maxWaitPeriodForRestoreCompletionInMinute*time.Minute, restoreJobProgressRetryTime*time.Minute, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restore %s", restoreName))
+		})
+
+		Step("Restore the backup taken with replace policy and move destination ns to no project while restoring", func() {
+			log.InfoD("Restore the backup taken with replace policy and move destination ns to no project while restoring")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreName = fmt.Sprintf("%s-replace-ns-move-to-no-project-%v", restoreNamePrefix, RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreOnRancherWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx, nil, projectUIDMapping, projectNameMapping, 2)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore- %s from backup %s", restoreName, backupName))
+			log.Infof("Move destination ns to no project while restore is going on")
+			err = Inst().S.(*rke.Rancher).RemoveNamespaceFromProject(restoreNamespaceList)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Moving namespaces %v from project %s to no project", restoreNamespaceList, destProjectList[1]))
+			log.Infof("Verifying if restore is successful after moving destination namespaces to no project")
+			err = restoreSuccessCheck(restoreName, orgID, maxWaitPeriodForRestoreCompletionInMinute*time.Minute, restoreJobProgressRetryTime*time.Minute, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restore %s", restoreName))
```

Comment:
Mention the names of all the namespaces and restores for all these log statements

Reply:
done